report/renderer/performance-category-renderer.js

```diff
@@ -219,41 +219,42 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
         .filter(audit => this._isPerformanceInsight(audit))
-        .filter(audit => !ReportUtils.showAsPassed(audit.result))
-        .sort((a, b) => {
-          // Sort audits by impact, prioritizing those with a higher overallImpact first,
-          // then falling back to linearImpact, guidance level and score.
-          const {
-            overallImpact: aOverallImpact,
-            overallLinearImpact: aOverallLinearImpact,
-          } = this.overallImpact(a, metricAudits);
-          const {
-            overallImpact: bOverallImpact,
-            overallLinearImpact: bOverallLinearImpact,
-          } = this.overallImpact(b, metricAudits);
-
-          const aGuidanceLevel = a.result.guidanceLevel || 1;
-          const bGuidanceLevel = b.result.guidanceLevel || 1;
-
-          if (aOverallImpact !== bOverallImpact) return bOverallImpact - aOverallImpact;
-
-          if (
-            aOverallImpact === 0 && bOverallImpact === 0 &&
-            aOverallLinearImpact !== bOverallLinearImpact
-          ) {
-            return bOverallLinearImpact - aOverallLinearImpact;
-          }
+        .filter(audit => !ReportUtils.showAsPassed(audit.result));
+    /** @type {Array<{auditRef:LH.ReportResult.AuditRef, overallImpact: number, overallLinearImpact: number, guidanceLevel: number}>} */
+    const auditImpacts = [];
+    diagnosticAudits.forEach(audit => {
+      const {
+        overallImpact: overallImpact,
+        overallLinearImpact: overallLinearImpact,
+      } = this.overallImpact(audit, metricAudits);
+      // eslint-disable-next-line max-len
+      auditImpacts.push({auditRef: audit, overallImpact, overallLinearImpact, guidanceLevel: audit.result.guidanceLevel || 1});
+    });
+
+    auditImpacts.sort((a, b) => {
+      // Sort audits by impact, prioritizing those with a higher overallImpact first,
+      // then falling back to linearImpact, guidance level and score.
+      if (a.overallImpact !== b.overallImpact) return b.overallImpact - a.overallImpact;
 
-          if (aGuidanceLevel !== bGuidanceLevel) return bGuidanceLevel - aGuidanceLevel;
+      if (
+        a.overallImpact === 0 && b.overallImpact === 0 &&
+        a.overallLinearImpact !== b.overallLinearImpact
+      ) {
+        return b.overallLinearImpact - a.overallLinearImpact;
+      }
+
+      if (a.guidanceLevel !== b.guidanceLevel) return b.guidanceLevel - a.guidanceLevel;
 
-          const scoreA = a.result.scoreDisplayMode === 'informative' ? 100 : Number(a.result.score);
-          const scoreB = b.result.scoreDisplayMode === 'informative' ? 100 : Number(b.result.score);
-          return scoreA - scoreB;
-        });
+      // eslint-disable-next-line max-len
+      const scoreA = a.auditRef.result.scoreDisplayMode === 'informative' ? 100 : Number(a.auditRef.result.score);
+      // eslint-disable-next-line max-len
+      const scoreB = b.auditRef.result.scoreDisplayMode === 'informative' ? 100 : Number(b.auditRef.result.score);
```

Comment:
nit: I think it would be better to obey the eslint rule here. Could reformat to something like:

```js
const scoreA = a.auditRef.result.scoreDisplayMode === 'informative'
  ? 100
  : Number(a.auditRef.result.score);
```